ooni/oonicli.py

```diff
@@ -315,3 +329,231 @@ def start():
         return d
 
     return start()
+
+# this variant version of runWithDirector splits the process in two,
+# allowing a single director instance to be reused with multiple decks
+# Some clever refactoring could reunite this with its original, but I'm
+# wary of disturbing the original.  In particular, the start_tor variable
+# was problematic.
+
+def runWithDaemonDirector(logging=True, start_tor=True, check_incoherences=True):
+    """
+    Instance the director, parse command line options and start an ooniprobe
+    test!
+    """
+    global_options = parseOptions()
+    config.global_options = global_options
+    config.set_paths()
+    config.initialize_ooni_home()
+    try:
+        config.read_config_file(check_incoherences=check_incoherences)
+    except errors.ConfigFileIncoherent:
+        sys.exit(6)
+
+    if global_options['verbose']:
+        config.advanced.debug = True
+
+    if not start_tor:
+        config.advanced.start_tor = False
+
+    if logging:
+        log.start(global_options['logfile'])
+
+    if config.privacy.includepcap:
+        if hasRawSocketPermission():
+            from ooni.utils.txscapy import ScapyFactory
+            config.scapyFactory = ScapyFactory(config.advanced.interface)
+        else:
+            log.err("Insufficient Privileges to capture packets."
+                    " See ooniprobe.conf privacy.includepcap")
+            sys.exit(2)
+
+    director = Director()
+
+    if global_options.get('annotations') is not None:
+        annotations = {}
+        for annotation in global_options["annotations"].split(","):
+            pair = annotation.split(":")
+            if len(pair) == 2:
+                key = pair[0].strip()
+                value = pair[1].strip()
+                annotations[key] = value
+            else:
+                log.err("Invalid annotation: %s" % annotation)
+                sys.exit(1)
+        global_options["annotations"] = annotations
+
+    if global_options['no-collector']:
+        log.msg("Not reporting using a collector")
+        global_options['collector'] = None
+        start_tor = False
+    else:
+        start_tor = True
+
+    def createDeck(url=None,filename=None):
```

Comment:
This function should probably go inside the `ooni.deck` module and perhaps be a method or be alternative parameters to the constructor of `ooni.deck`.